builtin/mainmenu/dlg_config_world.lua

```diff
@@ -101,18 +101,24 @@ local function enable_mod(this, toset)
 		-- game mods can't be enabled or disabled
 	elseif not mod.is_modpack then
 		if toset == nil then
-			mod.enabled = not mod.enabled
-		else
-			mod.enabled = toset
+			toset = not mod.enabled
+		end
+		for i = 1, #modmgr.mods_by_name[mod.name] do
+			local mod_to_set = modmgr.mods_by_name[mod.name][i]
+			mod_to_set.enabled = mod_to_set.modpack == mod.modpack and toset
 		end
 	else
 		local list = this.data.list:get_raw_list()
 		for i=1,#list,1 do
-			if list[i].modpack == mod.name then
+			local this_mod = list[i]
+			if this_mod.modpack == mod.name then
```

Comment:
Couldn't you do this in a 1-dimensional loop?
Something like
```Lua
for i, this_mod in pairs(list) do
	if this_mod.modpack ~= mod.name then
		this_mod.enabled = false
	else
		this_mod.enabled = (toset == nil) and not this_mod.enabled or toset
	end
end
```

Reply:
I could get rid of mods_by_name altogether, by adding another loop like this in the branch previous to this one. It will waste more cycles, because it will iterate through the whole list every time a single mod is enabled or disabled, but it will save in memory and complexity. What's preferred?

Reply:
I guess less complexity is preferred. This is not critical code anyway.

Reply:
Sorry, no, it is right as it is now. When activating a modpack, every mod in the modpack needs to be enabled, and every other mod with the same name as each of those in the modpack needs to be disabled. While iterating, whether a mod with the same name as the current one exists in the modpack can't be known until it is seen. That was the main reason for having `mods_by_name`. At best, two passes could be performed, one to collect the names of mods in the modpack and the other to disable conflicting names.